File: BinarySearch.c
#include <stdio.h>

int main()
{
    int nums[] = {2, 6, 9, 22, 43, 67, 99};
    int target = 67;
    int start = 0;
    int mid = 0;
    int end = sizeof(nums);

    while (start <= end)
    {
        mid = start + (end - start) / 2;

        if (nums[mid] == target)
        {
            printf("Found\n");
            return 0;
        }
        else if (target < nums[mid])
        {
            end = mid - 1;
        }
        else 
        {
            start = mid + 1;
        }
    }
    printf("Not found\n");
    return 0;
}

File: BinarySearch.py
from typing import List

data = [2, 6, 9, 22, 43, 67, 99]
target = 8
# comment 

def binary_search(data: List[int], target: int) -> bool:
    start = 0
    end = len(data) - 1

    while start <= end:
        mid = start + (end - start) // 2

        if target == data[mid]:
            print("Found")
            return True
        elif data[mid] < target:
            start = mid + 1
        else:
            end = mid - 1
    
    print("Not found")
    return False


def main():
    binary_search(data, target)

if __name__ == '__main__':
    main()
